src/ac_training_lab/cobot280pi/server.py

```diff
@@ -0,0 +1,246 @@
+	logger.info(f"running command control/angle with {args}")
+	try:
+		cobot.send_angles(**args)
+		return {"success": True}
+	except Exception as e:
+		logger.critical(f"control angle error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+def handle_control_coords(args, cobot):
+	logger.info(f"running command control/coord with {args}")
+	try:
+		cobot.send_coords(**args)
+		return {"success": True}
+	except Exception as e:
+		logger.critical(f"control coords error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+def handle_query_angles(args, cobot):
+	logger.info(f"running command query/angle with {args}")
+	try:
+		angles = cobot.get_angles()
+		if angles is None or len(angles) < 6:
+			raise Exception("could not read angle")
+		return {"success": True, "angles": angles}
+	except Exception as e:
+		logger.critical(f"query angle error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+def handle_query_coords(args, cobot):
+	logger.info(f"running command query/coord with {args}")
+	try:
+		coords = cobot.get_coords()
+		if coords is None or len(coords) < 6:
+			raise Exception("could not read coord")
+		return {"success": True, "coords": coords}
+	except Exception as e:
+		logger.critical(f"query coord error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+def handle_query_gripper(args, cobot):
+	logger.info(f"running command query/coord with {args}")
+	try:
+		gripper_pos = cobot.get_gripper_value()
+		return {"success": True, "position": gripper_pos}
+	except Exception as e:
+		logger.critical(f"query gripper error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+def handle_query_camera(args):
+	logger.info(f"running command query/camera with {args}")
+	try:
+		if not cliargs.debug:
+			webcam = cv2.VideoCapture(0)
+			_, frame = webcam.read()
+			webcam.release()
+			img = Image.fromarray(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+		else:
+			img = cobot.get_camera(**args)
+		compressed_bytes = io.BytesIO()
+		img.save(compressed_bytes, format="JPEG", quality=args["quality"])
+		compressed_bytes.seek(0)
+		byte_str = base64.b64encode(compressed_bytes.read()).decode("utf-8")
+		return {"success": True, "image": byte_str}
+	except Exception as e:
+		logger.critical(f"query camera error: {str(e)}")
+		return {"success": False, "error_msg": str(e)}
+
+
+# server logic
+task_queue = Queue()
+user_queue = Queue()
+user_queue_index = {}
+
+client = paho.Client(client_id="", userdata=None, protocol=paho.MQTTv5)
+client.tls_set(tls_version=paho.ssl.PROTOCOL_TLS)
+client.username_pw_set(HIVEMQ_USERNAME, HIVEMQ_PASSWORD)
+client.connect(HIVEMQ_HOST, DEVICE_PORT)
+
+# publishes a json str to specified response_topic, returns only when publish is complete
+def publish_message(
+	response_topic: str,
+	payload: dict[str, Any], 
+):
+	payload_str = json.dumps(payload)
+	pub_handle = client.publish(response_topic, qos=2, payload=payload_str)
+	# pub_handle.wait_for_publish()
```

Comment:
Could you mention why this is commented?

Reply:
https://github.com/eclipse-paho/paho.mqtt.python/issues/439#issuecomment-565514393

I think I ran into a problem where I needed to start the loop before using this

Reply:
fixed in the latest push, just forgot to uncomment